develop/index-best-practice.md

```diff
@@ -5,3 +5,126 @@ title: Best Practices for Indexing
 <!-- markdownlint-disable MD029 -->
 
 # Best Practices for Indexing
+
+This section introduces some best practices for creating and using index in TiDB.
+
+## Before you begin
+
+This section will use the `books` table in the [bookshop](/develop/bookshop-schema-design.md) database as an example.
+
+```sql
+CREATE TABLE `books` (
+  `id` bigint(20) AUTO_RANDOM NOT NULL,
+  `title` varchar(100) NOT NULL,
+  `type` enum('Magazine', 'Novel', 'Life', 'Arts', 'Comics', 'Education & Reference', 'Humanities & Social Sciences', 'Science & Technology', 'Kids', 'Sports') NOT NULL,
+  `published_at` datetime NOT NULL,
+  `stock` int(11) DEFAULT '0',
+  `price` decimal(15,2) DEFAULT '0.0',
+  PRIMARY KEY (`id`) CLUSTERED
+) DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin;
+```
+
+## Create Index Best Practices
+
+1. Create combined index of all the columns of data you need to use. This optimization technique is called [covering index optimization](https://docs.pingcap.com/tidb/stable/explain-indexes#indexreader). `covering index optimization` will allow TiDB to get all required data for the query directly on the index data, which can greatly improve performance.
+
+2. Avoid creating secondary index that you don't need. Useful secondary indexes can speed up queries, but be aware that adding an index has side effects. Every time you add an index, an additional key-value is added when you insert a row of data, so the more indexes you have, the slower you write, and the more space it takes up. In addition, too many indexes can affect optimizer runtime, and inappropriate indexes can mislead the optimizer. So more indexes aren't always better.
+
+3. Create an appropriate index based on business characteristics. In principle, you need to create indexes on the columns you need to use in your query to improve performance. The following situations are suitable for creating an index:
+ - Columns with a high degree of distinction can significantly reduce the number of filtered rows. For example, it is recommended to create an index on the column of the person's ID number, but not on the column of the person's gender.
+ - If you have multiple search conditions, you can use combined index. Note that columns with equivalent conditions need to be placed in front of the combined index
+ Here is an example. Assuming the query is `select* from t where c1 = 10 and c2 = 100 and c3 > 10`, then consider creating a combined index `Index cidx (c1, c2, c3) `, so that index prefix scan can be performed according to query conditions.
+ 
+ 4. Use meaningful secondary index name, and we recommend that you follow your company's or organization's table naming conventions. If your company or organization does not have an appropriate naming convention, refer to [Index Naming Specification](/develop/object-naming-guidelines.md)。
```

Comment:
```suggestion
3. Create an appropriate index based on business characteristics. In principle, you need to create indexes on the columns you need to use in your query to improve performance. The following situations are suitable for creating an index:

- Columns with a high degree of distinction can significantly reduce the number of filtered rows. For example, it is recommended to create an index on the column of the person's ID number, but not on the column of the person's gender.
- If you have multiple search conditions, you can use combined index. Note that columns with equivalent conditions need to be placed in front of the combined index.
Here is an example. Assuming the query is `select* from t where c1 = 10 and c2 = 100 and c3 > 10`, then consider creating a combined index `Index cidx (c1, c2, c3)`, so that index prefix scan can be performed according to query conditions.

4. Use meaningful secondary index name, and we recommend that you follow your company's or organization's table naming conventions. If your company or organization does not have an appropriate naming convention, refer to [Index Naming Specification](/develop/object-naming-guidelines.md).
```